tests/util/dev-node.ts

```diff
@@ -0,0 +1,125 @@
+import tcpPortUsed from "tcp-port-used";
+import { spawn, ChildProcess } from "child_process";
+import { BINARY_PATH, DISPLAY_LOG, MOONBEAM_LOG, SPAWNING_TIME } from "./constants";
+
+export async function findAvailablePorts() {
+  const availablePorts = await Promise.all(
+    [null, null, null].map(async (_, index) => {
+      let selectedPort = 0;
+      let port = 1024 + index * 20000 + (process.pid % 20000);
+      let endingPort = 65535;
+      while (!selectedPort && port < endingPort) {
+        const inUse = await tcpPortUsed.check(port, "127.0.0.1");
+        if (!inUse) {
+          selectedPort = port;
+        }
+        port++;
+      }
+      if (!selectedPort) {
+        throw new Error(`No available port`);
+      }
+      return selectedPort;
+    })
+  );
+
+  return {
+    p2pPort: availablePorts[0],
+    rpcPort: availablePorts[1],
+    wsPort: availablePorts[2],
+  };
+}
+
+// Stores if the node has already started.
+// It is used when a test file contains multiple describeDevMoonbeam. Those are
+// executed within the same PID and so would generate a race condition if started
+// at the same time.
+let nodeStarted = false;
+
+// This will start a moonbeam dev node, only 1 at a time (check every 100ms).
+// This will prevent race condition on the findAvailablePorts which uses the PID of the process
+export async function startMoonbeamDevNode(): Promise<{
+  p2pPort: number;
+  rpcPort: number;
+  wsPort: number;
+  runningNode: ChildProcess;
+}> {
+  while (nodeStarted) {
+    // Wait 100ms to see if the node is free
+    await new Promise((resolve) => {
+      setTimeout(resolve, 100);
+    });
+  }
+  nodeStarted = true;
+  const { p2pPort, rpcPort, wsPort } = await findAvailablePorts();
+
+  // console.log("Using ports", { p2pPort, rpcPort, wsPort });
+  const cmd = BINARY_PATH;
+  const args = [
+    `--execution=Native`, // Faster execution using native
```

Comment:
Faster is great. But our docs tell our node operators to run `--execution=wasm` so in that sense we're testing a different thing that we expect the users to use.